src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -202,293 +202,247 @@ async function importDotnetJs(resourceLoader: WebAssemblyResourceLoader): Promis
+  }
 
-      // Fetch the assemblies and PDBs in the background, telling Mono to wait until they are loaded
-      // Mono requires the assembly filenames to have a '.dll' extension, so supply such names regardless
-      // of the extensions in the URLs. This allows loading assemblies with arbitrary filenames.
-      assembliesBeingLoaded.forEach(r => addResourceAsAssembly(r, changeExtension(r.name, '.dll')));
-      pdbsBeingLoaded.forEach(r => addResourceAsAssembly(r, r.name));
+  if (resourceLoader.bootConfig.icuDataMode === ICUDataMode.Sharded) {
+    environmentVariables['__BLAZOR_SHARDED_ICU'] = '1';
+  }
 
-      Blazor._internal.dotNetCriticalError = printErr;
-      Blazor._internal.loadLazyAssembly = loadLazyAssembly;
+  if (resourceLoader.startOptions.applicationCulture) {
+    // If a culture is specified via start options use that to initialize the Emscripten \  .NET culture.
+    environmentVariables['LANG'] = `${resourceLoader.startOptions.applicationCulture}.UTF-8`;
+  }
 
-      // Wire-up callbacks for satellite assemblies. Blazor will call these as part of the application
-      // startup sequence to load satellite assemblies for the application's culture.
-      Blazor._internal.loadSatelliteAssemblies = loadSatelliteAssemblies;
-    };
+  if (resourceLoader.bootConfig.aspnetCoreBrowserTools) {
+    // See https://github.com/dotnet/aspnetcore/issues/37357#issuecomment-941237000
+    environmentVariables['__ASPNETCORE_BROWSER_TOOLS'] = resourceLoader.bootConfig.aspnetCoreBrowserTools;
+  }
 
-    async function loadSatelliteAssemblies(culturesToLoad: string[], loader: (wrapper: { dll: Uint8Array }) => void): Promise<void> {
-      const satelliteResources = resourceLoader.bootConfig.resources.satelliteResources;
-      if (!satelliteResources) {
-        return;
-      }
-      await Promise.all(culturesToLoad!
-        .filter(culture => satelliteResources.hasOwnProperty(culture))
-        .map(culture => resourceLoader.loadResources(satelliteResources[culture], fileName => `_framework/${fileName}`, 'assembly'))
-        .reduce((previous, next) => previous.concat(next), new Array<LoadingResource>())
-        .map(async resource => {
-          const response = await resource.response;
-          const bytes = await response.arrayBuffer();
-          const wrapper = { dll: new Uint8Array(bytes) };
-          loader(wrapper);
-        }));
-    }
 
-    async function loadLazyAssembly(assemblyNameToLoad: string): Promise<{ dll: Uint8Array, pdb: Uint8Array | null }> {
-      const lazyAssemblies = resources.lazyAssembly;
-      if (!lazyAssemblies) {
-        throw new Error("No assemblies have been marked as lazy-loadable. Use the 'BlazorWebAssemblyLazyLoad' item group in your project file to enable lazy loading an assembly.");
-      }
+  const moduleConfig = (window['Module'] || {}) as typeof Module;
+  // TODO (moduleConfig as any).preloadPlugins = []; // why do we need this ?
```

Comment:
It was added here: https://github.com/dotnet/aspnetcore/commit/4138b3a049331dc527b8b379e63d0de131934d16 and since then I could not find any trace of referencing it.

Reply:
@SteveSandersonMS could we delete `preloadPlugins`  property ?

Reply:
As far as I know yes, it can be deleted (assuming nothing obvious breaks), since we don't appear to be using it.

It goes right back to the original version of this code, so perhaps way back then something else needed the value to exist and not be null. Things have changed so many times since then.